[BUG] Magisk Manager with random package name crashes after installing LSPosed V1.5.2

<!--
如果你遇到了卡开机的问题，请确保在提交错误报告之前禁用了所有无关的 Xposed 模块和 Magisk 模块
If you encountered boot loop, please make sure you have disabled all unrelated Xposed and Magisk modules before submit an issue

无论如何，请使用英文标题
In any case, the title should be in English
-->

**什么情况/What happened?**
升级到LSPosed V1.5.2之后，以前用随机包名隐藏的Magisk Manager无法打开，随后安装了原版包名的Magisk Manager可以打开，但依然无法打开随机包名的版本。
<!--
如无法开机/模块无效等
Such as bootloop, module not loaded, etc
-->

**Xposed模块列表/Xposed Module List**
微信增强
QNotified
验证码提取器(Xposed)
知了
自动记账
<!--
如果为模块问题，请先确认出问题的具体是哪个模块。另外我们无法解决模块自身的问题，请直接联系模块作者。
If a module not working, please narrow down which specific module not working.
Mostly we can't help with module-specific issue, please contact their author first.
-->

**Magisk模块列表/Magisk Module List**
Riru
Riru - LSPosed
小米10应用全屏

**版本信息/Versions**
Android: 11

Magisk: 23.0

Riru: v26.1.1.r500.45d2706e83

LSPosed: v1.5.2 (5964)

**相关日志/Related Logs**

<!--
为了帮助我们定位问题，请使用 debug 版 LSPosed 模块抓取日志
Please capture log with debug version, otherwise it won't help us locating the issue
-->
```
[ 2021-09-01T17:35:51.819    10651:  6646:  6646 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6646
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:52.049    10651:  6694:  6694 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6694
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:52.247    10651:  6736:  6736 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6736
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:52.440    10651:  6773:  6773 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6773
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:52.637    10651:  6820:  6820 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6820
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:52.836    10651:  6887:  6887 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6887
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:53.036    10651:  6943:  6943 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6943
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:53.232    10651:  6993:  6993 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 6993
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:53.441    10651:  7044:  7044 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 7044
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:53.644    10651:  7093:  7093 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 7093
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
[ 2021-09-01T17:35:53.868    10651:  7181:  7181 E/AndroidRuntime  ] FATAL EXCEPTION: main
Process: rwhkydqhpc.pr, PID: 7181
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{rwhkydqhpc.pr/iG.tSb}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3699)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2135)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8057)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)' on a null object reference
	at w.HU.F(Unknown Source:1)
	at w.HU.instantiateActivity(Unknown Source:1)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1254)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3443)
	... 11 more
```

--- a/core/src/main/java/org/lsposed/lspd/service/ServiceManager.java
+++ b/core/src/main/java/org/lsposed/lspd/service/ServiceManager.java
@@ -94,6 +94,11 @@ public class ServiceManager {
 
         DdmHandleAppName.setAppName("lspd", 0);
 
+        // get config before package service is started
+        // otherwise getInstance will trigger module/scope cache
+        var configManager = ConfigManager.getInstance();
+        // --- DO NOT call ConfigManager.getInstance later!!! ---
+
         waitSystemService("package");
         waitSystemService("activity");
         waitSystemService(Context.USER_SERVICE);
@@ -122,7 +127,7 @@ public class ServiceManager {
         });
 
         // Force logging on boot, now let's see if we need to stop logging
-        if (!ConfigManager.getInstance().verboseLog()) {
+        if (!configManager.verboseLog()) {
             logcatService.stopVerbose();
         }
 

